File: app/src/main/java/org/tvheadend/tvhclient/data/entity/ServerProfile.java
package org.tvheadend.tvhclient.data.entity;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Index;
import androidx.room.PrimaryKey;

@Entity(tableName = "server_profiles", indices = {@Index(value = {"id"}, unique = true)})
public class ServerProfile {

    @PrimaryKey(autoGenerate = true)
    private int id;
    @ColumnInfo(name = "connection_id")
    private int connectionId;
    @ColumnInfo(name = "is_enabled")
    private boolean isEnabled = false;
    private String name;
    private String uuid;
    private String comment;
    private String type;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getConnectionId() {
        return connectionId;
    }

    public void setConnectionId(int connectionId) {
        this.connectionId = connectionId;
    }

    public boolean isEnabled() {
        return isEnabled;
    }

    public void setEnabled(boolean enabled) {
        isEnabled = enabled;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getUuid() {
        return uuid;
    }

    public void setUuid(String uuid) {
        this.uuid = uuid;
    }

    public String getComment() {
        return comment;
    }

    public void setComment(String comment) {
        this.comment = comment;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }
}

File: app/src/main/java/org/tvheadend/tvhclient/domain/entity/ServerProfile.kt
package org.tvheadend.tvhclient.domain.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(tableName = "server_profiles", indices = [Index(value = ["id"], unique = true)])
data class ServerProfile(

        @PrimaryKey(autoGenerate = true)
        var id: Int = 0,
        @ColumnInfo(name = "connection_id")
        var connectionId: Int = 0,
        @ColumnInfo(name = "is_enabled")
        var isEnabled: Boolean = false,
        var name: String? = null,
        var uuid: String? = null,
        var comment: String? = null,
        var type: String? = null
)
